specs/_features/eip7547/beacon-chain.md

```diff
@@ -0,0 +1,273 @@
+
+    return True
+```
+
+### Block processing
+
+#### Execution payload
+
+##### Modified `process_execution_payload`
+
+Assert validity of inclusion list summary with `verify_inclusion_list_summary_signature`.
+
+```python
+def process_execution_payload(state: BeaconState, body: BeaconBlockBody, execution_engine: ExecutionEngine) -> None:
+    payload = body.execution_payload
+
+    # Verify consistency of the parent hash with respect to the previous execution payload header
+    assert payload.parent_hash == state.latest_execution_payload_header.block_hash
+    # Verify prev_randao
+    assert payload.prev_randao == get_randao_mix(state, get_current_epoch(state))
+    # Verify timestamp
+    assert payload.timestamp == compute_timestamp_at_slot(state, state.slot)
+    # [New in EIP7547] Verify previous proposer signature on inclusion list summary
+    assert verify_inclusion_list_summary_signature(state, payload.previous_inclusion_list_summary)
+
+    # Verify commitments are under limit
+    assert len(body.blob_kzg_commitments) <= MAX_BLOBS_PER_BLOCK
+
+    # Verify the execution payload is valid
+    versioned_hashes = [kzg_commitment_to_versioned_hash(commitment) for commitment in body.blob_kzg_commitments]
+    assert execution_engine.verify_and_notify_new_payload(
+        NewPayloadRequest(
+            execution_payload=payload,
+            versioned_hashes=versioned_hashes,
+            parent_beacon_block_root=state.latest_block_header.parent_root,
+        )
+    )
+
+    # Cache execution payload header
+    state.latest_execution_payload_header = ExecutionPayloadHeader(
+        parent_hash=payload.parent_hash,
+        fee_recipient=payload.fee_recipient,
+        state_root=payload.state_root,
+        receipts_root=payload.receipts_root,
+        logs_bloom=payload.logs_bloom,
+        prev_randao=payload.prev_randao,
+        block_number=payload.block_number,
+        gas_limit=payload.gas_limit,
+        gas_used=payload.gas_used,
+        timestamp=payload.timestamp,
+        extra_data=payload.extra_data,
+        base_fee_per_gas=payload.base_fee_per_gas,
+        block_hash=payload.block_hash,
+        transactions_root=hash_tree_root(payload.transactions),
+        withdrawals_root=hash_tree_root(payload.withdrawals),
+        blob_gas_used=payload.blob_gas_used,
+        excess_blob_gas=payload.excess_blob_gas,
+        previous_inclusion_list_summary_root=hash_tree_root(payload.previous_inclusion_list_summary)  # [New in EIP7547]
+    )
+```
+
+##### New `verify_inclusion_list_summary_signature`
+
+```python
+def verify_inclusion_list_summary_signature(state: BeaconState,
+                                            inclusion_list_summary: SignedInclusionListSummary) -> bool:
+    signing_root = compute_signing_root(
+        inclusion_list_summary.message,
+        get_domain(state, DOMAIN_INCLUSION_LIST_SUMMARY),
+    )
+    il_proposer = state.validators[inclusion_list_summary.message.proposer_index]
```

Comment:
I believe we need to verify that `inclusion_list_summary` matches the previous summary. Therefore, we need the beacon state to track the previous proposer index. Or else an evil next-slot proposer could modify the summary to use its own index or something silly